### PR TITLE
pidgin: Only set Tcl/Tk paths on MacOS

### PR DESCRIPTION
**Pull request by dholm**

### CODE REVIEW

**Comment by sjackman** [comment]
This PR on its own is good.

CircleCI fails with…
```
checking for XML::Parser... configure: error: XML::Parser perl module is required for intltool
```
https://circleci.com/gh/Linuxbrew/homebrew-core/19633

The easiest way to resolve this issue is to add…
```ruby
depends_on "perl" unless OS.mac?
```

@iMichka Sound good?

**Comment by dholm** [comment]
@sjackman That is fine by me, fixed!

**Comment by sjackman** [comment]
```
checking for tclConfig.sh... no
configure: error: 
Tcl development headers not found.
Use --disable-tcl if you do not need Tcl scripting support.
```
Should we `--disable-tcl` on Linux, or add `tcl-tk` as a dependency?

**Comment by dholm** [comment]
@sjackman Agreed, I'll disable it.

**Comment by iMichka** [comment]
@dholm Sorry for the delay. Could you please rebase this? Upstream updated the options list, so there is a small conflict to solve.

By the way, do you maybe know why the test is failing?

**Comment by dholm** [comment]
@iMichka Done!

**Comment by sjackman** [comment]
```
==> brew test pidgin --verbose
Testing pidgin
==> /home/linuxbrew/.linuxbrew/Cellar/pidgin/2.13.0_2/bin/finch --version
.
Last 150 lines from /home/linuxbrew/.cache/Homebrew/Logs/pidgin/test.01.finch:
2018-09-18 12:20:09 +0000

/home/linuxbrew/.linuxbrew/Cellar/pidgin/2.13.0_2/bin/finch
--version

Failed to execute: /home/linuxbrew/.linuxbrew/Cellar/pidgin/2.13.0_2/bin/finch
Error: pidgin: failed
==> FAILED
```
https://circleci.com/gh/Linuxbrew/homebrew-core/20383

It looks as though `brew test pidgin` fails when it runs `finch --version`. Can you reproduce this error on your local machine?

**Comment by iMichka** [comment]
```
configure: error: 

Finch will not be built. You need to install ncursesw (or ncurses) and its development headers.

==> Formula
Path: /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pidgin.rb
```

The `ncurses` dependency is missing too.

File: Formula/pidgin.rb [hunk 1]
```diff
@@ -22,7 +22,9 @@ class Pidgin < Formula
   depends_on "libgcrypt"
   depends_on "libidn"
   depends_on "libotr"
+  depends_on "ncurses"
   depends_on "pango"
+  depends_on "perl" unless OS.mac?
```
Comment (iMichka) [inline Formula/pidgin.rb]: Please regroup both in a single block:
```
unless OS.mac?
  depends_on "ncurses"
  depends_on "perl"
end
```

File: Formula/pidgin.rb [hunk 2]
```diff
@@ -45,12 +47,18 @@ def install
       --disable-gtkspell
       --disable-meanwhile
       --disable-vv
+      --enable-consoleui
```
Comment (iMichka) [inline Formula/pidgin.rb]: Can you add this flag below in the linux block. We are adding this flag only for Linux.

File: Formula/pidgin.rb (hunk 1 above)
Comment (dholm) [reply Formula/pidgin.rb]: Fixed

**Comment by dholm** [comment]
@iMichka Fixed!

**Comment by dholm** [comment]
@iMichka Fixed!

**Comment by dholm** [comment]
@iMichka Fixed!

File: Formula/pidgin.rb [hunk 3]
```diff
@@ -23,6 +23,10 @@ class Pidgin < Formula
   depends_on "libidn"
   depends_on "libotr"
   depends_on "pango"
+  unless OS.mac?
+    depends_on "ncurses"
+    depends_on "perl" unless OS.mac?
```
Comment (sjackman) [inline Formula/pidgin.rb]: Please remove this redundant `unless OS.mac?`

Comment (dholm) [reply Formula/pidgin.rb]: Oops, fixed!

**Comment by sjackman** [comment]
```
Finch will not be built. You need to install ncursesw (or ncurses) and its development headers.
```
https://circleci.com/gh/Linuxbrew/homebrew-core/21511

Try adding `depends_on "ncurses" unless OS.mac?```

**Comment by iMichka** [comment]
`depends_on "ncurses"` was added, but it seems to not be enough ... :(

**Pull request closed without merging**